tsdb/index/tsi1/tag_block.go

```diff
@@ -539,6 +543,13 @@ func (enc *TagBlockEncoder) EncodeValue(value []byte, deleted bool, seriesIDs []
 		return fmt.Errorf("zero length tag value not allowed")
 	}
 
+	// Validate that keys are in-order.
```

Comment:
How often will we call `EncodeValue`? I only ask as this `bytes.Compare` call could be an expensive way to test the encoder is working properly. Or is returning an error a "normal path" for this encoder?

Reply:
It's called once for every tag value in the index. I think the cost should be relatively small compared to the merging and writing. We can remove it if it becomes a performance bottleneck but I think it'd be good to keep it in the short-term at least to ensure we don't store unsorted data.